src/raftstore/store/store.rs

```diff
@@ -1456,6 +1616,241 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         }
     }
 
+    fn register_merge_check_tick(&self, event_loop: &mut EventLoop<Self>) {
+        if let Err(e) = register_timer(
+            event_loop,
+            Tick::CheckMerge,
+            self.cfg.merge_check_tick_interval.as_millis(),
+        ) {
+            error!("{} register split region check tick err: {:?}", self.tag, e);
+        };
+    }
+
+    fn get_merge_peer(&self, tag: &str, target_region: &metapb::Region) -> Result<Option<&Peer>> {
+        let region_id = target_region.get_id();
+        if let Some(p) = self.region_peers.get(&region_id) {
+            if p.region().get_region_epoch().get_conf_ver()
+                > target_region.get_region_epoch().get_conf_ver()
+            {
+                return Err(box_err!(
+                    "target region changed {:?} -> {:?}",
+                    target_region,
+                    p.region()
+                ));
+            }
+            return Ok(Some(p));
+        }
+
+        let state_key = keys::region_state_key(region_id);
+        let state: RegionLocalState = match self.kv_engine().get_msg_cf(CF_RAFT, &state_key) {
+            Err(e) => {
+                error!(
+                    "{} failed to load region state of {}, ignore: {}",
+                    tag, region_id, e
+                );
+                return Ok(None);
+            }
+            Ok(None) => {
+                info!(
+                    "{} seems to merge into a new replica of region {}, let's wait.",
+                    tag, region_id
+                );
+                return Ok(None);
+            }
+            Ok(Some(state)) => state,
+        };
+        if state.get_state() != PeerState::Tombstone {
+            info!("{} wait for region {} split.", tag, region_id);
```

Comment:
Why a region has `RegionLocalState` but not in `region_peers` must be spliting or tombstone?

Reply:
Because only split will write a new `RegionLocalState` but update `region_peers` asynchronously.

Reply:
Then is it possible to compare `region_local_state.get_region_epoch()` to `expect_epoch`? So we don't need to try it again in next round. And it can be more unify with comparing `expect_epoch` with `exist_epoch` from `region_peers`.

Reply:
Sure, it's more appropriate to do it in other pr.